tests/unit/remote/test_gdrive.py

```diff
@@ -0,0 +1,325 @@
+        "appDataFolder"
+    )  # don't do extra call inside the .root property
+    return gdrive
+
+
+@pytest.fixture(autouse=True)
+def no_requests(monkeypatch):
+    req_mock = mock.Mock(return_value=Response(""))
+    monkeypatch.setattr("requests.sessions.Session.request", req_mock)
+    return req_mock
+
+
+def _url(url):
+    return RemoteGDrive.GOOGLEAPIS_BASE_URL + url
+
+
+@pytest.fixture(autouse=True)
+def fake_creds(monkeypatch):
+
+    creds = google.oauth2.credentials.Credentials(
+        token="MOCK_token",
+        refresh_token="MOCK_refresh_token",
+        token_uri="MOCK_token_uri",
+        client_id="MOCK_client_id",
+        client_secret="MOCK_client_secret",
+        scopes=["MOCK_scopes"],
+    )
+    creds.expiry = datetime.now() + timedelta(days=1)
+
+    mocked_flow = mock.Mock()
+    mocked_flow.run_console.return_value = creds
+    mocked_flow.run_local_server.return_value = creds
+
+    monkeypatch.setattr(
+        InstalledAppFlow,
+        "from_client_secrets_file",
+        classmethod(lambda *args, **kwargs: mocked_flow),
+    )
+
+    monkeypatch.setattr(
+        OAuth2, "_get_creds_id", mock.Mock(return_value="test")
+    )
+
+
+@pytest.fixture(autouse=True)
+def no_refresh(monkeypatch):
+    monkeypatch.setattr(
+        "google.oauth2.credentials.Credentials.refresh", mock.Mock()
+    )
+
+
+@pytest.fixture()
+def makedirs(monkeypatch):
+    mocked = mock.Mock(return_value="FOLDER_ID")
+    monkeypatch.setattr(RemoteGDrive, "makedirs", mocked)
+    return mocked
+
+
+def test_init_drive(gdrive):
+    assert gdrive.url == GDRIVE_URL
+    assert gdrive.oauth2.scopes == ["https://www.googleapis.com/auth/drive"]
+    assert gdrive.space == RemoteGDrive.SPACE_DRIVE
+
+
+def test_init_appfolder(gdrive_appfolder):
+    assert gdrive_appfolder.url == GDRIVE_APPFOLDER_URL
+    assert gdrive_appfolder.oauth2.scopes == [
+        "https://www.googleapis.com/auth/drive.appdata"
+    ]
+    assert gdrive_appfolder.space == RemoteGDrive.SPACE_APPDATA
+
+
+def test_init_folder_id(repo):
+    url = "gdrive://FOLDER_ID/subpath"
+    remote = RemoteGDrive(repo, {"url": url})
+    assert remote.url == url
+    assert remote.oauth2.scopes == ["https://www.googleapis.com/auth/drive"]
+    assert remote.space == "drive"
+
+
+def test_path_info(repo):
+    remote = RemoteGDrive(repo, {"url": "gdrive://root"})
+    assert remote.path_info.root == "root"
+
+
+def test_root(repo, makedirs):
+    remote = RemoteGDrive(repo, {"url": "gdrive://root/folder"})
+    assert remote.root == "FOLDER_ID"
+    makedirs.assert_called_once_with("root", "folder")
+
+
+def test_root_appfolder(repo, makedirs):
+    remote = RemoteGDrive(repo, {"url": "gdrive://dvc/folder"})
+    assert remote.root == "FOLDER_ID"
+    makedirs.assert_called_once_with("appDataFolder", "folder")
+
+
+def test_get_session(gdrive, no_requests):
+    session = gdrive.oauth2.get_session()
+    # XXX(ei-grad): it actually shouldn't work with domains different
```

Comment:
So should it work or shouldn't? If shouldn't, why test with non-google domain?

Reply:
The `google-auth-oauthlib` implementation uses wrong method to add its headers. We can switch to our own, I even tried and have it implemented somewhere in stash, but gdrive unit tests became too complex in requests mocking.

Reply:
So it actually works. We can change domain to googleapis and remove this comment, but then this knowlege would be lost. Should we create an issue for this?

Reply:
Yeah, let's remove it. Not sure what is the issue here, tbh :slightly_smiling_face: